core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java

```diff
@@ -69,23 +82,109 @@ public void testGetNonexistentView() throws Exception {
+    assertThat(view.getEnd().getSeconds()).isAtLeast(view.getStart().getSeconds());
+    List<DistributionAggregation> distributionAggregations = view.getDistributionAggregations();
+    assertThat(distributionAggregations).hasSize(1);
+    DistributionAggregation distributionAggregation = distributionAggregations.get(0);
+    verifyDistributionAggregation(distributionAggregation,
+        RpcConstants.RPC_MILLIS_BUCKET_BOUNDARIES.size() + 1, 4,100.0,25.0,10.0,40.0,1);
+    List<Tag> tags = distributionAggregation.getTags();
+    assertThat(tags.get(0).getKey()).isEqualTo(tagKey);
+    assertThat(tags.get(0).getValue()).isEqualTo(tagValue1);
+  }
+
+  @Test
+  public void testRecordMultipleTagValues() {
+    statsManager.registerView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
+    statsManager.record(new StatsContextImpl(oneTag),
+        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0));
+    statsManager.record(new StatsContextImpl(anotherTag),
+        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 30.0));
+    statsManager.record(new StatsContextImpl(anotherTag),
+        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 50.0));
+
+    DistributionView view =
+        (DistributionView) statsManager.getView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
+    List<DistributionAggregation> distributionAggregations = view.getDistributionAggregations();
+    assertThat(distributionAggregations).hasSize(2);
+    DistributionAggregation distributionAggregation1 = distributionAggregations.get(0);
+    DistributionAggregation distributionAggregation2 = distributionAggregations.get(1);
+
+    // The order of distributionAggregation1 and distributionAggregation2 could be random
+    if (distributionAggregation1.getCount() == 2) {
+      DistributionAggregation distributionAggregation = distributionAggregation1;
+      distributionAggregation1 = distributionAggregation2;
+      distributionAggregation2 = distributionAggregation;
+    }
+
+    verifyDistributionAggregation(distributionAggregation1,
+        RpcConstants.RPC_MILLIS_BUCKET_BOUNDARIES.size() + 1, 1,10.0,10.0,10.0,10.0,1);
+    verifyDistributionAggregation(distributionAggregation2,
+        RpcConstants.RPC_MILLIS_BUCKET_BOUNDARIES.size() + 1, 2,80.0,40.0,30.0,50.0,1);
+    assertThat(distributionAggregation1.getTags().get(0).getKey()).isEqualTo(tagKey);
+    assertThat(distributionAggregation1.getTags().get(0).getValue()).isEqualTo(tagValue1);
+    assertThat(distributionAggregation2.getTags().get(0).getKey()).isEqualTo(tagKey);
+    assertThat(distributionAggregation2.getTags().get(0).getValue()).isEqualTo(tagValue2);
+  }
+
+  private static void verifyDistributionAggregation(
+      DistributionAggregation distributionAggregation, int bucketCountsSize, int count, double sum,
+      double mean, double min, double max, int tagsSize) {
+    assertThat(distributionAggregation.getBucketCounts().size()).isEqualTo(bucketCountsSize);
```

Comment:
Sorry, I didn't finish reviewing the changes.

We don't omit the trailing buckets now, but we could change it.  Here is the comment: https://github.com/google/instrumentation-java/blob/55f477ece335dad5dc5ca949870f0f216d369931/core/src/main/java/com/google/instrumentation/stats/DistributionAggregation.java#L106 .  Since you added the check for the non-zero buckets, I think you could just remove this check.